core/model/modx/modmenu.class.php

```diff
@@ -101,6 +101,8 @@ public function getSubMenus($start = '') {
         $c->where(array(
             'modMenu.parent' => $start,
         ));
+        // make sure installer is always on top
```

Comment:
Let's give users a choice and make it optional (simply on/off) in 2.x.

Reply:
If we make a system setting, then it should be enabled by default.

Reply:
Excellent! Great for in-the-minority folks like me who want it elsewhere.

Reply:
And we should fix the issue with the menu indexes in 3.x and look into the other great suggestions!

Reply:
Updating an extra resets not only the menuindex. It resets also the parent of menu entries. Both should stay the same.

Reply:
It also resets the permissions.

Reply:
Ok, that should be moved to a different issue.